sdk/azidentity/managed_identity_client.go

```diff
@@ -171,14 +173,16 @@ func (c *managedIdentityClient) createIMDSAuthRequest(ctx context.Context, clien
 	q := request.URL.Query()
 	q.Add("api-version", c.imdsAPIVersion)
 	q.Add("resource", strings.Join(scopes, " "))
-	if clientID != "" {
-		q.Add(qpClientID, clientID)
+	if c.id == ResourceID {
```

Comment:
This logic should be its own func to avoid the duplication.  Might also be safer to implement as a `switch` with a `default` clause to guard against future values being added and the logic not properly updated.

Reply:
I'm a little hesitant to abstract this into a func to avoid duplication because I dont know that all of the ID kinds are supported in every MSI environment. I checked that resource ID is supported for those that I added it to in this PR. Therefore this check is a per environment functionality that needs to be added. 
For instance App Service also supports `principal_id` which we may add to IDKind in the future, but azure VMs dont. Instead Azure VMs support `object_id`.

Reply:
Makes sense, leave as-is then.